crates/util/src/paths.rs

```diff
@@ -382,6 +382,8 @@ pub fn compare_paths(
                 });
                 if !ordering.is_eq() {
                     return ordering;
+                } else if !component_a.as_os_str().cmp(component_b.as_os_str()).is_eq() {
```

Comment:
This looks quite odd, it says that we do
```if !ordering.is_eq() && a != b {
  b.cmp(&a)
}
```

Isn't this `ordering.then_with(|| component_b.as_os_str().cmp(component_a.as_os_str()))`?

I find it also a bit confusing that we do `num_and_remainder_a.cmp(&num_and_remainder_b)` (a with b) comparison above, and (b with a) comparison here, changing the direction comparison.
Let's add a comment explaining this?

Also curious if we can avoid `.as_os_str()` call at all, as there's a `PartialOrd` on the `Ordering` itself?

Reply:
Yeah, `.reverse()` + some small comment on top of the new comparison section seems to be the best option.

Reply:
We can indeed place the thing into the `NumericPrefixWithSuffix`, but maybe not in every branch but below the match, so that it looks like `match .. {}.then_with(|| ...reverse())` + maybe have more tests there instead (?).

But that all does not matter that much, current place is also fine.